code/modules/projectiles/guns/beam_rifle.dm

```diff
@@ -472,7 +472,7 @@
 		if(original && (original.layer>=2.75) || ismob(original))
```

Comment:
huh it actually is the code that is responsible for hitting the actual target object